master/buildbot/util/githubapiservice.py

```diff
@@ -0,0 +1,245 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import datetime
+import pprint
+import textwrap
+import time
+from collections import OrderedDict
+from collections import namedtuple
+
+import jinja2
+import jwt
+
+from twisted.internet import defer
+
+from buildbot.util.httpclientservice import HTTPClientService
+from buildbot.util.logger import Logger
+from buildbot.util.service import SharedService
+
+log = Logger()
+
+
+class GithubApiService(SharedService):
```

Comment:
why is it not a subclass of HTTPClientService?

Reply:
I believe I saw this kind of usage back when I initially wrote this some months ago by looking around at other services.

I'm happy to subclass it if it's also a shared service and gets the job done.